pkg/volume/util/atomic_writer.go

```diff
@@ -445,18 +444,10 @@ func (w *AtomicWriter) writePayloadToDir(payload map[string]FileProjection, dir
 		if fileProjection.FsUser == nil {
 			continue
 		}
-<<<<<<< HEAD
-		if isNotWindows {
-			if err := os.Chown(fullPath, int(*fileProjection.FsUser), -1); err != nil {
-				klog.Errorf("%s: unable to change file %s with owner %v: %v", w.logContext, fullPath, int(*fileProjection.FsUser), err)
-				return err
-			}
-=======
```

Comment:
This is part of one of our downstream patches and we can't remove it: https://github.com/bertinatto/kubernetes/commit/f12da041f22e

Reply:
We need to bring these changes back, please check my comment above.

Reply:
Are you sure? I checked the release-4.18 branch, and this is how it looks there right now: https://github.com/openshift/kubernetes/blob/acb9630c20d2fa0d11bad4c38c7b28e2db9d818b/pkg/volume/util/atomic_writer.go#L448

Reply:
Yeah, we dropped that carry patch in 4.18 because upstream fixed the bug in k8s 1.31. We don't have the fix in 4.17, so we should keep it in `release-4.17`.

Our patch changed the code to only run `os.Chown()` on non-Windows machines. However, in k8s 1.31 (4.18) upstream fixed it by introducing `w.chown()` instead. This method will just log a warning on Windows ([see](https://github.com/openshift/kubernetes/blob/master/pkg/volume/util/atomic_writer_unsupported.go#L31)).

Reply:
> We don't have the fix in 4.17, so we should keep it in release-4.17.

Sorry, that's wrong. I didn't realize that the upstream was [backported to 1.30 as well](https://github.com/kubernetes/kubernetes/pull/129507).